core/modules/node/node.module

```diff
@@ -2136,6 +2135,28 @@ function node_page_default() {
 }
 
 /**
+ * Helper function to determine callback for /node path.
+ * @return string
+ *  which callback function to use for the /node path.
+ */
+function _node_path_or_not() {
+  $front = config_get('system.core', 'site_frontpage');
+  if ($front != 'node') {
+    return 'node_custom_front_page';
+  }
+  else {
+    return 'node_page_default';
+  }
+}
+
+/**
+ * If using a custom front page redirect the /node path there.
```

Comment:
...nit: add a comma after the "front page". Other wise it reads "front page redirect" (redirect verb vs noun).